src/vanilla/utils/unwrap.ts

```diff
@@ -35,6 +38,11 @@ export function unwrap<Value, Args extends unknown[], Result, PendingValue>(
   anAtom: WritableAtom<Value, Args, Result> | Atom<Value>,
   fallback: (prev?: Awaited<Value>) => PendingValue = defaultFallback as never,
 ) {
+  const stableFallback =
+    fallback === defaultFallback
+      ? fallback
+      : getStableFallback(fallback, anAtom)
```

Comment:
This doesn't work for certain cases if we want dynamic callbacks.
Let's educate users to define callbacks outside component or use `useCallback`.

Reply:
Alright, I can drop the commit. But could you elaborate just so I understand, what would be the use case? And is it something you would prefer to support over the one in my example above?

Thinking further, this definitely needs to be documented as a breaking change - a Jotai eslint plugin could be useful to disallow inline callbacks here if that's what you mean 🤔

Reply:
Not a specific use case in mind, but the function can be anything and `prev => prev ?? DEFAULT` is only one use case.

It's not a change as I see it. Not only `unwrap`, but also `useAtomCallback` always expect stable selectors. It's kind of surprising inline callbacks worked in previous versions as it's not intended.

That said, because `atomWithDefault((get) => get(unwrap(anAsyncAtom, (prev) => prev ?? DEFAULT)))` worked previously by chance, it's better to document it for explicitly. Would you like to open another PR for doc improvement?

My recommendation is to separate it into two atoms:
```ts
const atomA = unwrap(anAsyncAtom, (prev) => prev ?? getSomeDefaultValueThatMayChange());
const atomB = atomWithDefault((get) => get(atomA));
```

Reply:
That's fair, thank you. I can make another PR to update the documentation. I'll drop the commit now.